simplecpp.cpp

```diff
@@ -1752,8 +1753,15 @@ static bool realFileName(const std::string &f, std::string *result)
 
     // Lookup filename or foldername on file system
     WIN32_FIND_DATAA FindFileData;
-    HANDLE hFind = FindFirstFileA(&buf[0], &FindFileData);
-    if (hFind == INVALID_HANDLE_VALUE)
+	HANDLE hFind = INVALID_HANDLE_VALUE;
+	// only fetch once
+	static BOOL bIsWindows7OrGreater = IsWindows7OrGreater();
+	if (bIsWindows7OrGreater)
+		hFind = FindFirstFileExA(&buf[0], FindExInfoBasic, &FindFileData, FindExSearchNameMatch, NULL, 0);
```

Comment:
I don't understand this code. Why do you only want to call this function in newer windows? Did it exist previously but did not work "properly"?

Reply:
This was because previously i also used FIND_FIRST_EX_LARGE_FETCH which is only supported by >Win7 systems